### DISCUSSION TITLE
Tree Generator: Add def imsquare(n) to makevegdems.py

### DISCUSSION THREAD

**Discussion by rarygit** [UMEP-dev/UMEP#281] [Q&A] [answered]
At the moment I am using a grid of trees to mimic canopy cover and shading from a square pergola structure.

The aim is to begin exploring canopy topology optimisation where tree structures are not feasible (lack of space) or the community is risk-averse to falling branches.

Although the Tmrt and surface temperature response from the shading is informative, using imcircle(n) as a proxy for imsquare(n) is unsatisfactory.

Would imsquare(n) be feasible to implement in Tree Generator? 
Would there be a better approach to implement pergola-like vegetation canopy structures in Tree Generator?
e.g. "Canopy Generator"

**Comment by biglimp** [comment, marked as answer]
Yes, you could try to replace circle with square.

**Reply by rarygit** [reply]
Thanks for replying, I will try this substitution and model individual pergolas. 

But my preference is to have both options for circle and square geometry functions in the makevegdems.py script. 

And one step further into the unknown:  I would like to determine, i.e. set, the height of the canopy trunk space for each side of the square.

The objective is to have shade along the south and east or south-southeast "faces" of the square (facing the sun, e.g. trunk space is 10% of tree canopy height) and set the height of the other two faces of the square (trunk space) as 70% canopy height. 

For the "sitting person", I would like diffuse sunlight entering from two faces; and direct sunlight shaded from the other two faces.

Conceptually as shown in the images attached below, but using vegetation canopy rather than the wooden lattices and rafters shown in the images (source: taken google.com/images).   

![two_sides_shaded](https://user-images.githubusercontent.com/16574504/119821116-7b9a3980-bef2-11eb-85b7-1f12c7fd72c9.jpeg)
![two_sides_shaded2](https://user-images.githubusercontent.com/16574504/119821119-7c32d000-bef2-11eb-9084-8f7c4288a156.jpeg)
![two_sides_shaded3](https://user-images.githubusercontent.com/16574504/119821120-7ccb6680-bef2-11eb-870b-cae40f488bf8.jpeg)

**Reply by biglimp** [reply]
Hm, in this case I would manipulate the data outside of the TreeGenerator and do it directly in QGIS by crerating two thin but high trees (the walls) and one flat squared tree with very high trunk zone.

**Reply by rarygit** [reply]
Thanks, I will try this idea.

**Reply by rarygit** [reply]
I experimented by adapting your second suggestion: outside of TreeGenerator, create one thin tree (3.5 m H, 'vegetated lattice', one side) with a low trunk zone (0.5 m H) and one flat squared tree (3.5 m H, 'vegetated rafters') with very high trunk zone (2.5 m H). So this pergola has been constructed using the DSM Generator in UMEP.

It was an interesting exercise to create the pergola. 
Some tweaking with the raster output to get what I wanted in terms of the pergola's cdsm and tdsm files
I am still cogitating the method, but I like the prospect of combining "rafter and lattice" shading effects

FYI, as attached, any comments, suggestions would be welcome regarding the vegetated pergola structure results. 
Especially the POI.txt output. I have compared this to simulations using trees and other tree-based pergola designs
The met file is a slice of the timeseries for an 11 day heat wave across central europe, 2015-08-05 to 2015-08-11

 
[2015_218_219_Pergola.zip](https://github.com/UMEP-dev/UMEP/files/6559059/2015_218_219_Pergola.zip)

![EDO_Heatwave_20150807](https://user-images.githubusercontent.com/16574504/119956788-bc9d5700-bfa1-11eb-9ab2-e53ec6393af2.jpg)

**Comment by rarygit** [comment]
@biglimp: we can mark this as answered.
